#!/usr/bin/env python
PACKAGE = "ltu_actor_route_obstacle"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("norm_x",   double_t,    0, "norm x", 0.0, -2, 2)
gen.add("norm_y",   double_t,    0, "norm y", 0.0, -2, 2)
gen.add("norm_z",   double_t,    0, "norm z", 1.0, -2, 2)
gen.add("theta",    double_t,    0, "theta" , -0.28, -2, 2)

#---- Front Close Region
gen.add("front_close_originX",   double_t,    0, "front close: originX", 1.0, -10.0, 50.0)
gen.add("front_close_originY",   double_t,    0, "front close: originY", -0.5, -10.0, 10.0)
gen.add("front_close_width",   double_t,    0, "front close: width", 1.0, 0, 10.0)
gen.add("front_close_length",   double_t,    0, "front close: length", 7.5, 0, 30)
gen.add("front_close_z_min",   double_t,    0, "front close: z_min", -0.6, -10, 10)
gen.add("front_close_z_max",   double_t,    0, "front close: z_max", 0.5, -10, 10)

#---- Front Far Region
gen.add("front_far_originX",   double_t,    0, "front far: originX", 4.0, -10.0, 50.0)
gen.add("front_far_originY",   double_t,    0, "front far: originY", -0.5, -10.0, 10.0)
gen.add("front_far_width",   double_t,    0, "front far: width", 1.0, 0, 10.0)
gen.add("front_far_length",   double_t,    0, "front far: length", 4.0, 0, 30)
gen.add("front_far_z_min",   double_t,    0, "front_far: z_min", -1.25, -10, 10)
gen.add("front_far_z_max",   double_t,    0, "front_far: z_max", 0.0, -10, 10)

#---- Left Close Region
gen.add("left_close_originX",   double_t,    0, "left close: originX", 1.0, -10.0, 50.0)
gen.add("left_close_originY",   double_t,    0, "left close: originY", -1.5, -10.0, 10.0)
gen.add("left_close_width",   double_t,    0, "left close: width", 1.0, 0, 10.0)
gen.add("left_close_length",   double_t,    0, "left close: length", 3.0, 0, 30)
gen.add("left_close_z_min",   double_t,    0, "left_close: z_min", -1.25, -10, 10)
gen.add("left_close_z_max",   double_t,    0, "left_close: z_max", 0.0, -10, 10)

#---- Left Far Region, needs updates
gen.add("left_far_originX",   double_t,    0, "left far: originX", 4.0, -10.0, 50.0)
gen.add("left_far_originY",   double_t,    0, "left far: originY", -0.5, -10.0, 10.0)
gen.add("left_far_width",   double_t,    0, "left far: width", 1.0, 0, 10.0)
gen.add("left_far_length",   double_t,    0, "left far: length", 4.0, 0, 30)
gen.add("left_far_z_min",   double_t,    0, "left_far: z_min", -1.25, -10, 10)
gen.add("left_far_z_max",   double_t,    0, "left_far: z_max", 0.0, -10, 10)

#---- Right Close Region
gen.add("right_close_originX",   double_t,    0, "right close: originX", 1.0, -10.0, 50.0)
gen.add("right_close_originY",   double_t,    0, "right close: originY", 0.5, -10.0, 10.0)
gen.add("right_close_width",   double_t,    0, "right close: width", 1.0, 0, 10.0)
gen.add("right_close_length",   double_t,    0, "right close: length", 3.0, 0, 30)
gen.add("right_close_z_min",   double_t,    0, "right_close: z_min", -1.25, -10, 10)
gen.add("right_close_z_max",   double_t,    0, "right_close: z_max", 0.0, -10, 10)

#---- Right Far Region, needs updates
gen.add("right_far_originX",   double_t,    0, "right far: originX", 4.0, -10.0, 50.0)
gen.add("right_far_originY",   double_t,    0, "right far: originY", -0.5, -10.0, 10.0)
gen.add("right_far_width",   double_t,    0, "right far: width", 1.0, 0, 10.0)
gen.add("right_far_length",   double_t,    0, "right far: length", 4.0, 0, 30)
gen.add("right_far_z_min",   double_t,    0, "right_far: z_min", -1.25, -10, 10)
gen.add("right_far_z_max",   double_t,    0, "right_far: z_max", 0.0, -10, 10)

exit(gen.generate(PACKAGE, "actor_obstacle", "Region"))
